Update readme

nmorduch opened the pull request:
Some information is uncertain but should still help with troubleshooting. I didn't think it was worth me taking the time to figure out what the actual answer was for "maybe x needs y", because having that information is still an improvement for troubleshooting. There certainly may be errors in here; feel free to correct them :)

- Add files for nvm & pyenv to mark Node & Python versions. 
- Update setup instructions.
  - Add information about using a db dump
  - Add notes about using AWS
  - Clarify some instructions about files from Google Drive
  - Add instructions for venv as an alternative to virtualenv
  - Improve formatting

tomdyson commented:
Nice!

In general I think you can be a bit more opinionated: developers for this repo will either already be confident with virtualenvs etc or won't need instructions for two options.

And I'd fix the capitalisation throughout :) s3 -> S3, wagtail -> Wagtail, postgres -> Postgres etc.

nmorduch commented:
Agreed; for now, though, I just want to make sure the information is up there, and cleaning it up can come later :)

kevinhowbrook commented on README.md:
```diff
@@ -1,116 +1,170 @@
-- [Python 3 and pip](https://docs.python-guide.org/starting/install3/osx/)
-- [Node.js](https://medium.com/@kkostov/how-to-install-node-and-npm-on-macos-using-homebrew-708e2c3877bd)
-- [virtualenv](http://virtualenvwrapper.readthedocs.org/en/latest/index.html) (`pip install virtualenv`)
+- [Python 3](https://docs.python-guide.org/starting/install3/osx/), version noted in [.python-version](.python-version)
```
I think we should still leave pip in this line

nmorduch replied:
Looks I got mixed up a bit about when pip is automatically included. venv installs it by default in recent versions of Python, and I was thinking that means later versions of Python had it by default. Will update.

kevinhowbrook commented on README.md:
```diff
@@ -1,116 +1,170 @@
+- [virtualenv](http://virtualenvwrapper.readthedocs.org/en/latest/index.html) (`pip install virtualenv`) or [venv](https://docs.python.org/3/library/venv.html)
 - [postgres](http://exponential.io/blog/2015/02/21/install-postgresql-on-mac-os-x-via-brew/)
-- [aws-cli](https://aws.amazon.com/cli/) (`pip install awscli`)
+- [aws-cli](https://aws.amazon.com/cli/) (`pip install awscli` and you'll need an account at [aws.amazon.com](https://aws.amazon.com/) with access to S3)
```
I think something a bit more descriptive here like:

[aws-cli](https://aws.amazon.com/cli/) (`pip install awscli`). You'll need an account at [aws.amazon.com](https://aws.amazon.com/) with an 'iam' user set up which is configured in a later step on this README. [More information about awscli configuration](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration)

nmorduch replied:
will do :)

kevinhowbrook commented on README.md:
````diff
@@ -1,116 +1,170 @@
 ```
 
-- If you'd like your local instance to include images:
+#### If you're using a database dump
````
This section is useful, it helped us get started, but I'd much rather we work from updated working fixtures. So once https://github.com/newamericafoundation/newamerica-cms/issues/1374 is resolved, we'll remove this section but it's useful for now.

nmorduch commented:
Addressed your comments @kevinhowbrook, ready for rereview. I realized that I needed the node version thing merged, so this became a higher priority for me.
